addons/purchase_requisition/models/purchase.py

```diff
@@ -258,6 +258,16 @@ def get_tender_best_lines(self):
             best_price_unit_ids.update(lines.ids)
         return list(best_price_ids), list(best_date_ids), list(best_price_unit_ids)
 
+    def _prepare_grouped_data(self, rfq):
+        match_fields = super()._prepare_grouped_data(rfq)
+        return match_fields + (rfq.requisition_id.id,)
+
+    def _check_alternative_purchase_orders(self, rfqs, oldest_rfq):
```

Comment:
make it a method on oldest_rfq
```suggestion
    def _merge_alternative_po(self, rfqs):
```
and call it 
```python
oldest_rfq._merge_alternative_po(rfqs)
```